src/classes/AppState.js

```diff
@@ -5,50 +5,62 @@ export class AppState {
     this.is_auth = false;
     this.roles = [];
   }
-
+  
   getUserID() {
     return this.user._id;
   }
-
-  getUser(uid) {
+  
+  async getUser(uid) {
     console.log('appState getUser');
     if (this.getUserID() !== undefined) {
       console.log('appState returning already set user');
       return new Promise((resolve) => {
         resolve(this.user);
       });
     }
+    await fetch;
     console.log('appState getting new user');
     return this.httpClient.fetch('/user/' + uid)
     .then(response => response.json())
     .then(data => {
       let user = data;
       this.setUser(user);
-      return this.user;
+      this.checkUserRole();
+      //return this.user;
     });
   }
-
+  
+  checkUserRole(){
+    if (this.user.userType !== 'Developer'){
+      this.setRoles([toLowerCase(this.user.userType)]);
+    } else {
+      this.setRoles(['charity', 'volunteer', 'developer', 'reader', 'librarian']);
+      //this.app.router.navigate('dashboard/developer');
+    }
+  }
+  
   setUser(input) {
     console.log('appState setUser');
     console.log(this.user);
     this.user = input;
   }
-
+  
```

Comment:
Really need to get editorconfig setup, so it isn't putting these extra spaces on empty lines.